Neos.Neos/Classes/Domain/Context/ContentStream/EditorContentStreamZookeeper.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Neos\Neos\Domain\Context\ContentStream;
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Domain\Context\Workspace\Command\CreateWorkspace;
+use Neos\ContentRepository\Domain\Context\Workspace\Command\RebaseWorkspace;
+use Neos\ContentRepository\Domain\Context\Workspace\WorkspaceCommandHandler;
+use Neos\ContentRepository\Domain\Projection\Workspace\Workspace;
+use Neos\ContentRepository\Domain\Projection\Workspace\WorkspaceFinder;
+use Neos\ContentRepository\Domain\ValueObject\ContentStreamIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\UserIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\WorkspaceDescription;
+use Neos\ContentRepository\Domain\ValueObject\WorkspaceName;
+use Neos\ContentRepository\Domain\ValueObject\WorkspaceTitle;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\Security\Authentication;
+use Neos\Flow\Security\Policy\PolicyService;
+use Neos\Flow\Security\Policy\Role;
+use Neos\Neos\Domain\Model\User;
+use Neos\Party\Domain\Service\PartyService;
+
+/**
+ * The service for keeping track of editors' content streams
+ *
+ * On authentication, workspaces may have to be created and content streams may have to be forked from live or rebased from older ones
+ */
+final class EditorContentStreamZookeeper
+{
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @Flow\Inject
+     * @var PartyService
+     */
+    protected $partyService;
+
+    /**
+     * @Flow\Inject
+     * @var PolicyService
+     */
+    protected $policyService;
+
+    /**
+     * @Flow\Inject
+     * @var WorkspaceFinder
+     */
+    protected $workspaceFinder;
+
+    /**
+     * @Flow\Inject
+     * @var WorkspaceCommandHandler
+     */
+    protected $workspaceCommandHandler;
+
+
+    /**
+     * @param Authentication\TokenInterface $token
```

Comment:
THis method is called on 
$dispatcher->connect(AuthenticationProviderManager::class, 'authenticatedToken', EditorContentStreamZookeeper::class, 'relayEditorAuthentication'); using signal/slot